File: backend/src/main/java/com/smartcharger/entity/Vehicle.java
package com.smartcharger.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 车辆信息实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "vehicle")
public class Vehicle extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "license_plate", nullable = false, length = 20)
    private String licensePlate;

    @Column(name = "brand", length = 50)
    private String brand;

    @Column(name = "model", length = 50)
    private String model;

    @Column(name = "battery_capacity", precision = 10, scale = 2)
    private BigDecimal batteryCapacity;

    @Column(name = "is_default", nullable = false)
    private Integer isDefault = 0;
}

File: frontend-client/src/types/vehicle.ts
// 车辆信息接口
export interface VehicleInfo {
  id: number
  userId: number
  licensePlate: string
  brand?: string
  model?: string
  batteryCapacity?: number
  isDefault: number  // 0=否，1=是
  createdTime: string
  updatedTime: string
}

// 添加车辆请求
export interface VehicleCreateRequest {
  licensePlate: string
  brand?: string
  model?: string
  batteryCapacity?: number
  isDefault?: number  // 0=否，1=是
}

// 更新车辆请求
export interface VehicleUpdateRequest {
  licensePlate?: string
  brand?: string
  model?: string
  batteryCapacity?: number
}

// 车牌号正则表达式
export const LICENSE_PLATE_PATTERN = /^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领][A-Z][A-HJ-NP-Z0-9]{4,5}[A-HJ-NP-Z0-9挂学警港澳]$/

// 车牌号验证函数
export function validateLicensePlate(licensePlate: string): boolean {
  return LICENSE_PLATE_PATTERN.test(licensePlate)
}
